devimint/src/util.rs

```diff
@@ -331,25 +332,37 @@ const DEFAULT_RETRIES: usize = 30;
 pub async fn poll<Fut, R>(
     name: &str,
     // TODO: this should be `Duration`, not number of retries --dpc
-    retries: impl Into<Option<usize>>,
+    timeout: impl Into<Option<Duration>>,
     f: impl Fn() -> Fut,
 ) -> Result<R>
 where
     Fut: Future<Output = Result<R, ControlFlow<anyhow::Error, anyhow::Error>>>,
 {
-    let retries = retries.into().unwrap_or(DEFAULT_RETRIES);
-    for i in 0.. {
+    let start = now();
+    let timeout = timeout.into().unwrap_or(DEFAULT_POLL_TIMEOUT);
+    for attempt in 0u64.. {
+        let attempt_start = now();
         match f().await {
             Ok(value) => return Ok(value),
             Err(ControlFlow::Break(err)) => {
                 return Err(err).with_context(|| format!("polling {name}"));
             }
-            Err(ControlFlow::Continue(err)) if i <= retries => {
-                debug!("polling {name} failed with: {err:?}, will retry... ({i}/{retries})");
-                task::sleep(Duration::from_secs(1)).await;
+            Err(ControlFlow::Continue(err))
+                if attempt_start
+                    .duration_since(start)
+                    .expect("time goes forward")
+                    < timeout =>
+            {
+                debug!(target: LOG_DEVIMINT, %attempt, %err, "Polling {name} failed, will retry...");
+                task::sleep(Duration::from_millis(attempt.min(1000))).await;
```

Comment:
`attempt.min(1000)` is pretty aggressive. Should we multiply attempt by a factor (100?) or did you want to sleep only a few millis?

Reply:
1s is not agressive, IMO. All these commands are supposed to wrap in a handful of seconds if everything is OK.

Reply:
Perhaps I'm misreading, but `Duration::from_millis(attempt.min(1000))` will wait 1 milli on the first attempt, then 2 until 1000 attempts.

Reply:
Exactly. :D

1 ms + time of running the command